build.gradle.kts

```diff
@@ -174,7 +174,7 @@ publishing {
 
 signing {
 	setRequired({
-		project.version != "unspecified"
+		project.version != "unspecified" && gradle.taskGraph.hasTask("publishToSonatype")
```

Comment:
i am not sure if the `project.version` check is needed. I would rather check for the other two project variables we are using `signingKey` and `signingPassword` - this should cover all the cases sufficiently.

Not sure about the taskgraph, i think this is maybe too limiting, to a specific target

Reply:
I followed the suggestion in the [Signing Plugin docs](https://docs.gradle.org/current/userguide/signing_plugin.html#sec:conditional_signing) to include the Gradle task graph. My assumption is that we only want to sign iff there's a project version _and_ we publish to Sonatype. This also seems to fix the JitPack build:

https://jitpack.io/com/github/beatngu13/junit-pioneer/e0bcc6c426/build.log

Regarding `signingKey` and `signingPassword`: consider the case where these variables are absent in the release build (for whatever reason). Since signing is skipped, the build would continue and attempt to upload to Sonatype, it wouldn't fail early.

Not sure about this, but we could let the build fail in the `signin` task by throwing an appropriate exception, if that makes sense?

Reply:
on the other hand, what if i want to sign my packages to test everything is running as expected without uploading to SonaType. In this case i would rather not have a task dependency, nor a project property - tough question.

I agree with your fail fast approach on releases, but i also tend to have builds work as expected locally. Tricky situation :D

Reply:
Let's add another condition:

```kotlin
project.hasProperty("enforceSigning")
		|| project.version != "unspecified" && gradle.taskGraph.hasTask("publishToSonatype")
```

Then do:

```text
$ ./gradlew sign -PenforceSigning
```

🙈

But yeah, it's indeed tricky. At the moment, I don't think I have a clear preference … you?

Reply:
me neither, as said i agree on both sides :D you got my approve and i am fine - important is that jitpack works again :) for the rest we do have time